pkg/deploy/controller/deployment/factory.go

```diff
@@ -110,6 +111,10 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 					return false
 				}
 				if retries.Count > 1 {
+					deployment, ok := obj.(*kapi.ReplicationController)
+					if ok {
+						eventRecorder.Eventf(deployment, kapi.EventTypeWarning, "FailedRetry", "About to stop retrying %s: %v", deployment.Name, err)
```

Comment:
Not all transient errors are about updating the deployment. Actually the only fatal error in the deployment controller is when the utility for generating a deployer pod will error out.... Should we add more error types and distinguish here?